upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -71,10 +82,22 @@ func (tf *TemplateFunctions) WellKnownServiceIP(id int) (net.IP, error) {
 	return nil, fmt.Errorf("Unexpected IP address type for ServiceClusterIPRange: %s", tf.cluster.Spec.ServiceClusterIPRange)
 }
 
+// This will define the available functions we can use in our YAML models
+// If we are trying to get a new function implemented it MUST
+// be defined here.
 func (tf *TemplateFunctions) AddTo(dest template.FuncMap) {
 	dest["EtcdClusterMemberTags"] = tf.EtcdClusterMemberTags
 	dest["SharedVPC"] = tf.SharedVPC
 
+	// Network topology definitions
+	dest["IsTopologyPublic"]  = tf.IsTopologyPublic
+	dest["IsTopologyPrivate"] = tf.IsTopologyPrivate
+	dest["IsTopologyPrivateMasters"] = tf.IsTopologyPrivateMasters
+	dest["WithBastion"] = tf.WithBastion
+	dest["GetBastionImageId"] = tf.GetBastionImageId
+	dest["GetBastionZone"] = tf.GetBastionZone
+	dest["GetELBName32"] = tf.GetELBName32
+
```

Comment:
We can block them together - as long as `GetELBName32` is outside of the block - that is not topology only